src/js/parsers/captions/vttparser.js

```diff
@@ -409,50 +379,59 @@ define(['parsers/captions/vttcue'], function(VTTCue) {
                                 // In case of an error ignore rest of the cue.
                                 self.cue = null;
                                 self.state = 'BADCUE';
-                                continue;
+                                break;
                             }
                             self.state = 'CUETEXT';
-                            continue;
+                            break;
                         case 'CUETEXT':
-                            var hasSubstring = line.indexOf('-->') !== -1;
+                            var hasSubstring = arrowRegex.test(line);
                             // 34 - If we have an empty line then report the cue.
                             // 35 - If we have the special substring '-->' then report the cue,
                             // but do not collect the line as we need to process the current
                             // one as a new cue.
                             if (!line || hasSubstring && (alreadyCollectedLine = true)) {
                                 // We are done parsing self cue.
                                 if (self.oncue) {
+                                    currentCueBatch += 1;
                                     self.oncue(self.cue);
                                 }
                                 self.cue = null;
                                 self.state = 'ID';
-                                continue;
+                                break;
                             }
                             if (self.cue.text) {
                                 self.cue.text += '\n';
                             }
                             self.cue.text += line;
-                            continue;
+                            break;
                         case 'BADCUE': // BADCUE
                             // 54-62 - Collect and discard the remaining cue.
                             if (!line) {
                                 self.state = 'ID';
                             }
-                            continue;
+                            break;
                     }
                 }
-            } catch (e) {
 
-                // If we are currently parsing a cue, report what we have.
-                if (self.state === 'CUETEXT' && self.cue && self.oncue) {
-                    self.oncue(self.cue);
+                currentCueBatch = 0;
+                if (self.buffer) {
+                    try {
+                        requestFrame(processBuffer);
```

Comment:
Does this need to be in a try catch block? The only code executed is `requestAnimationFrame` or `setTimeout`.

Any error parsing will fire asynchronously and no get caught.